src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -314,3 +321,49 @@ def github_comment_reactions():
             continue
 
         metrics.incr("github_pr_comment.comment_reactions.success")
+
+
+# TODO(adas): Change the client typing to allow for multiple SCM Integrations
+def safe_for_comment(
+    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
+) -> bool:
+    try:
+        pullrequest_resp = gh_client.get_pullrequest(
+            repo=repository.name, pull_number=pull_request.key
+        )
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "gh_rate_limited", "code": e.code},
+            )
+        elif e.code == 404:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "missing_gh_pull_request", "code": e.code},
+            )
+        else:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "unknown_api_error", "code": e.code},
+            )
+            logger.exception("github.open_pr_comment.unknown_api_error")
+        return False
+
+    safe_to_comment = True
+    if pullrequest_resp["state"] != "open":
```

Comment:
I don't remember but was there an enum we could use here for the state of a PR

Reply:
if you're talking about this, it doesn't have a string version but i can look into it https://github.com/getsentry/sentry/blob/master/src/sentry/models/pullrequest.py#L94

Reply:
ok makes sense